xm-commons-domainevent/src/main/java/com/icthh/xm/commons/domainevent/config/WebApiSource.java

```diff
@@ -9,15 +9,16 @@
 import com.icthh.xm.commons.security.XmAuthenticationContext;
 import com.icthh.xm.commons.security.XmAuthenticationContextHolder;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.stereotype.Component;
 import org.springframework.web.servlet.AsyncHandlerInterceptor;
-import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 @Slf4j
 @Component
+@ConditionalOnProperty(value = "application.domain-event.enabled", havingValue = "true")
```

Comment:
@sbaginskiy please be sure that in case we do not have `application.domain-event.enabled` property in config it will not fail spring context initialisation.

Reply:
@amedvedchuk yes, I made sure of it. I put this annotation because WebApiSource depends on DomainEventProviderFactory which depends on `application.domain-event.enabled` property. This caused problems with initialization if `application.domain-event.enabled` property is missing